image/image_linux.go

```diff
@@ -144,6 +171,36 @@ func installCloudConfig(rootDir, gadgetDir string) error {
 	return osutil.CopyFile(cloudConfig, dst, osutil.CopyFlagOverwrite)
 }
 
+func customizeImage(rootDir, defaultsDir string, custo *Customizations) error {
+	// customize with cloud-init user-data
+	if custo.CloudInitUserData != "" {
+		varCloudDir := filepath.Join(rootDir, "/var/lib/cloud/seed/nocloud-net")
+		if err := os.MkdirAll(varCloudDir, 0755); err != nil {
+			return err
+		}
+		if err := ioutil.WriteFile(filepath.Join(varCloudDir, "meta-data"), []byte("instance-id: nocloud-static\n"), 0644); err != nil {
+			return err
+		}
+		dst := filepath.Join(varCloudDir, "user-data")
+		if err := osutil.CopyFile(custo.CloudInitUserData, dst, osutil.CopyFlagOverwrite); err != nil {
```

Comment:
Should we verify that user-data is at least valid YAML? If not now then maybe a TODO would be applicable.

Reply:
u-i image itself wasn't doing that check, this is kind of a legacy feature at this point